transformer_engine/pytorch/module/layernorm_linear.py

```diff
@@ -369,7 +371,7 @@ def forward(
             out, _ = allreduce(out, tp_group)
 
         # [*, in_features] -> [*, out_features] except first dimension changes for SP
-        out = out.view(-1, *inp.shape[1:-1], out.shape[-1])
+        out = out.view(-1, *inp_shape[1:-1], out_features)
```

Comment:
Why can't we just create out with the right shape in the gemm call instead?

Reply:
I think it would be a good idea if the GEMM extension implicitly flattened all input tensors to 2D. I'd prefer only including Python changes in this PR and leaving that change to a follow-up PR.